File: HashMap/1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        // should be of same length
        if(word1.length()!=word2.length()) {
            return false;
        }

        // should have the same character 

        Map<Character, Integer> map1 = new HashMap<>();
        for(char c: word1.toCharArray()) {
            map1.put(c, map1.getOrDefault(c, 0)+1);
        }

        Map<Character, Integer> map2 = new HashMap<>();
        for(char c: word2.toCharArray()) {
            map2.put(c, map2.getOrDefault(c, 0)+1);
        }

        if(!map1.keySet().equals(map2.keySet())) {
            return false;
        }

        // sort order of the values

        List<Integer> arr1 = new ArrayList<>(map1.values());
        List<Integer> arr2 = new ArrayList<>(map2.values());

        Collections.sort(arr1);
        Collections.sort(arr2);

        return arr1.equals(arr2);



    }
}

File: Array/1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1)!=len(word2):
            return False
        count1 = Counter(word1)
        count2 = Counter(word2)

        values_count1 = Counter(count1.values())
        values_count2 = Counter(count2.values())


        return set(count1.values()) == set(count2.values()) and set(count1.keys()) == set(count2.keys()) and values_count1 == values_count2